server/routes/getExposureOverTime.ts

```diff
@@ -0,0 +1,146 @@
+  const { chain, stash } = req.query;
+  let { startDate, endDate } = req.query;
+
+  if (!chain) return next(new Error(Errors.ChainIdNotFound));
+  const chainInfo = await models.Chain.findOne({ where: { id: chain } });
+  if (!chainInfo) return next(new Error(Errors.InvalidChain));
+
+  // if start and end date isn't given, we set it for a year from now
+  if (typeof startDate === 'undefined' || typeof endDate === 'undefined') {
+    startDate = new Date();
+    startDate.setFullYear(startDate.getFullYear() - 1);
+    startDate = new Date(startDate);
+    endDate = new Date();
+  }
+
+  const where: any = {
+    '$ChainEventType.chain$': chain,
+  };
+
+  // if stash is given
+  // eslint-disable-next-line no-unused-expressions
+  stash && (where['$HistoricalValidatorStatistic.stash$'] = stash);
+
+  if (startDate && endDate) {
+    where.created_at = {
+      [Op.between]: [startDate, endDate]
+    };
+  }
+
+  stakeOverTime = await models.HistoricalValidatorStatistic.findAll({
+    where,
+    order: [
+      ['created_at', 'ASC']
+    ],
+    attributes: ['stash', 'exposure', 'block'],
+    include: [ { model: models.ChainEventType } ]
+  });
+};
+
+const getTotalStakeOverTime = async (models, req: Request, res: Response, next: NextFunction) => {
+  getStakeOverTime(models, req, res, next);
+  validators = {};
+
+  if (stakeOverTime) { if (!stakeOverTime.length) return []; } else { return []; }
+
+  stakeOverTime.forEach((stake) => {
+    const event_data: IEventData = stake.dataValues.event_data;
+    const key = event_data.stash.toString();
+    if (key in validators) {
+      if (validators[key].findIndex((element) => (element.stash.toString() === key)) === -1) {
+        validators[key].push([event_data.block_number, event_data.exposure.total]);
+      }
+    } else {
+      validators[key] = [event_data.block_number, event_data.exposure.total];
+    }
+  });
+
+  return res.json({ status: 'Success', result: { validators } });
+};
+
+const getOwnStakeOverTime = async (models, req: Request, res: Response, next: NextFunction) => {
+  getStakeOverTime(models, req, res, next);
+  validators = {};
+
+  if (stakeOverTime) { if (!stakeOverTime.length) return []; } else { return []; }
+
+  stakeOverTime.forEach((stake) => {
+    const event_data: IEventData = stake.dataValues.event_data;
+    const key = event_data.stash.toString();
+    if (key in validators) {
+      if (validators[key].findIndex((element) => (element.stash.toString() === key)) === -1) {
+        validators[key].push([event_data.block_number, event_data.exposure.own]);
+      }
+    } else {
+      validators[key] = [event_data.block_number, event_data.exposure.own];
+    }
+  });
+
+  return res.json({ status: 'Success', result: { validators } });
+};
+
+const getOtherStakeOverTime = async (models, req: Request, res: Response, next: NextFunction) => {
+  getStakeOverTime(models, req, res, next);
```

Comment:
no return value here

Reply:
I used a global variable, but yeah the return approach is preferable, I made the changes in next commit.